docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -97,103 +60,146 @@ appId: dev.expo.eastestsexample # This is an example app id. Replace it with you
-## Build your app and run E2E tests on EAS Build
+<Step label="5">
+## Create an E2E test workflow
+
+At the root of your project, create a **.eas/workflows** directory. Then add a YAML file for your E2E test workflow, such as **.eas/workflows/e2e-test-android.yml**.
+
+```yaml .eas/workflows/e2e-test-android.yml
+name: e2e-test-android
+
+on:
+  pull_request:
+    branches: ['*'] # Run the E2E test workflow on every pull request.
+jobs:
+  build_android_for_e2e:
+    type: build
+    params:
+      platform: android
+      profile: e2e-test # your eas build profile for E2E test
+
+  maestro_test:
+    needs: [build_android_for_e2e]
+    type: maestro
+    params:
+      build_id: ${{ needs.build_android_for_e2e.outputs.build_id }}
+      flow_path: ['.maestro/home.yml']
+```
 
-To execute a custom build using the `build-and-maestro-test` profile that will build your app and run the Maestro E2E tests afterward, run the following command:
+This workflow builds an **.apk** for Android using the `e2e-test` build profile from the previous step. Then it runs the **.maestro/home.yml** flow on the built APK.
 
-<Terminal cmd={['$ eas build --profile build-and-maestro-test']} />
+Here's an example of the same test workflow for iOS:
 
-<ContentSpotlight
-  alt="Logs of successful Maestro tests"
-  src="/static/images/eas-build/tests/03-logs.png"
-  className="max-w-[600px]"
-/>
+```yaml .eas/workflows/e2e-test.yml
+name: e2e-test-ios
 
-When the flow fails, any Maestro artifacts are automatically uploaded as build artifacts. This includes screenshots saved at `~/.maestro/tests` (the default destination). You can download them from the build page.
+on:
+  pull_request:
+    branches: ['*']
 
-<ContentSpotlight
-  alt="Failed Maestro tests artifacts"
-  src="/static/images/eas-build/tests/04-artifacts.png"
-  className="max-w-[600px]"
-/>
-</Step>
+jobs:
+  build_ios_for_e2e:
+    type: build
+    params:
+      platform: ios
+      profile: e2e-test # your eas build profile for E2E test
 
-## More
+  maestro_test:
+    needs: [build_ios_for_e2e]
+    type: maestro
+    params:
+      build_id: ${{ needs.build_ios_for_e2e.outputs.build_id }}
+      flow_path: ['.maestro/home.yml']
+```
 
-If you want to build more advanced custom builds workflows, see the [custom build schema reference](/custom-builds/schema/) for more information.
+Learn more about [Syntx for EAS Workflows](https://docs.eas.dev/workflows/syntax).
 
-To learn more about Maestro flows and how to write them, see the [Maestro documentation](https://maestro.mobile.dev).
+</Step>
 
-## Troubleshooting
+<Step label="6">
+## Run the E2E test workflow
 
-If you encounter the following error message when starting an Android Emulator for E2E tests on EAS Build:
+You can run the E2E test workflow in two ways:
 
-```text
-Failed to configure emulator: emulator with required ID not found.
+1. **Manually using the EAS CLI**
+
+```bash
+npx eas-cli@latest workflow:run .eas/workflows/e2e-test-android.yaml
```

Comment:
```suggestion
npx eas-cli@latest workflow:run .eas/workflows/e2e-test-android.yml
```

This should be "yml" instead of "yaml" considering the file name set in the previous step.